README.md

```diff
@@ -128,86 +91,24 @@ You can dynamically decide which transport to use, basing e.g. on the user-agent
 
 ### Basic usage
 
-Sessions are typed; the `T` type parameter in `SessionManager[T]` determines what data is stored in the session. 
-Basic types like `String`, `Int`, `Long`, `Float`, `Double` and `Map[String, String]` are supported out-of-the box. 
-Support for other types can be added by providing an implicit `SessionSerializer[T, String]`. For case classes, it's most 
-convenient to use a `MultiValueSessionSerializer[T]` which should convert the instance into a `String -> String` map 
-(nested types are not supported on purpose, as session data should be small & simple).
-
-Here we are creating a manager where the session content will be a single `Long` number:
+Sessions are typed. The `T` type parameter in `SessionManager[T]` (or `SessionManager<T>`) determines what data is stored in the session. 
+Basic types like `String`, `Int`, `Long`, `Float`, `Double` and `Map[String, String]` (`Map<String, String>`) are supported out-of-the box. 
+Support for other types can be added by providing an implicit `SessionSerializer[T, String]` (`SessionSerializer<T, String>`). For case classes, it's most 
+convenient to use a `MultiValueSessionSerializer[T]` or (`MultiValueSessionSerializer<T>`) which should convert the instance into a `String -> String` map 
+(nested types are not supported on purpose, as session data should be small & simple). Examples of `SessionSerializer` and `MultiValueSessionSerializer` 
+usage can be found [here](https://github.com/softwaremill/akka-http-session/blob/master/example/src/main/scala/com/softwaremill/example/serializers) for scala [here](https://github.com/softwaremill/akka-http-session/blob/master/example/src/main/java/com/softwaremill/example/serializers) for java. 
 
-````scala
-import com.softwaremill.session.{SessionConfig, SessionManager}
-
-val sessionConfig = SessionConfig.default("some_very_long_secret_and_random_string_some_very_long_secret_and_random_string")
-implicit val sessionManager = new SessionManager[Long](sessionConfig)
-````
+Sample code that illustrates how to create a session manager where the session content will be a single `Long` number can be found [here](https://github.com/softwaremill/akka-http-session/blob/master/example/src/main/scala/com/softwaremill/example/session/manager/MyScalaSessionManager.scala) for scala and [here](https://github.com/softwaremill/akka-http-session/blob/master/example/src/main/java/com/softwaremill/example/session/manager/MyJavaSessionManager.java) for java.
```

Comment:
This one could be like the following: `Here are code samples in [Scala](link) and [Java](link) illustrating ...`

What do you think?